CppCoreGuidelines.md

```diff
@@ -11165,6 +11165,16 @@ Use plain `{}`-initialization unless you specifically want to disable explicit c
         // ...
     }
 
+##### Exception
+
+For container-like types whose element type can be initialized from an instance of that very container
+type iteself, e.g. `std::any` or a template parameter `T` of a template type having a constructor accepting `T`,
+use `()` to avoid erroneously constructing a nested container:
+
+    vector<any> v{1,2,3};
+    vector<any> u{v}; // most likely unintended: u has 1 element of type any, which contains v
```

Comment:
I feel  `u{v}; // most likely unintended: u has 1 element` contradicts `v2{10}; // vector of 1 element with the value 10` a few lines above; constructing a 1-element vector holding v as its element is what curlies show the intent of.

Reply:
The "title" of the guidline is _Prefer the `{}`-initializer syntax_, which per se does not refer to initializer lists nor anything else that is discussed in the various examples.

The same holds for the **Reason** paragraph.

I think that's enough for the reader to interpret the guildeline as _"Ok, I'll blindly use `{}` for initializing anything until I get some compile-time or run-time error"_.

Furthermore, the example you mention is constrasting _list of elements_ vs _size_ for containers, whereas the case I'm making is about _list of elements_ vs _copy_, and the whole item does not make any mention as regards preferring `{}` vs `()` for copy constructing an entity, so in the absernce of an explicit reference to that, I think it's legit that the reader assumes that they should prefer `{}`, following the title of the guideline.

Reply:
Editors call: Thanks!

> I think that's enough for the reader to interpret the guildeline as _"Ok, I'll blindly use `{}` for initializing anything until I get some compile-time or run-time error"_.

The guideline starts with "Prefer" so we think that it doesn't say to blindly use braces. Also, `any` is a good example but a very special outlier type; `vector<int> x(1, 2)` is the stronger example and we think is covered in the first Exception. So we don't think a change is needed.